IridiumSBD Debug Message + Acknowlege fix

acfloria opened the pull request:
- Add a message with the status of the iridiumsbd driver. It is published at every loop of the driver if the status changed.

- Only send the acknowledge for the VEHICLE_CMD_CONTROL_HIGH_LATENCY command if the command is coming from external.

dagar commented:
Looks good. The check if you need to publish is a bit verbose. That makes me wonder if something similar could be pushed into the uORB layer itself (that already has the last message).

dagar commented:
Do you want to add iridiumsbd_status to logger?

acfloria commented:
I added it to the logger.

I guess this check is quite specific to the status messages. In general I think the messages would almost always change when published, for example sensor data, control inputs, estimator data, ...

acfloria commented on src/modules/commander/commander.cpp:
```diff
@@ -1056,7 +1056,10 @@ Commander::handle_command(vehicle_status_s *status_local,
 	case vehicle_command_s::VEHICLE_CMD_CONTROL_HIGH_LATENCY: {
 			// only send the acknowledge from the commander, the command actually is handled by each mavlink instance
-			cmd_result = vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED;
+			// only send the acknowledge if the command is received from an external source
```
The acknowledge is handled by the commander. The reason is that in that way transmitting multiple acknowledges is avoided.

dagar replied:
Sure, but you also lose the ability to even potentially acknowledge in a meaningful way. 

Even if you only sent the acknowledge in the mavlink task that's in iridium mode at least you'd get no response when Iridium is disabled instead of a meaningless acceptance.

acfloria replied:
Which is also not ideal because then QGC will be resending the command multiple times. What I could do here is to check if any high_latency telemetry exists and decide based on that if the command is accepted or denied. Would that be acceptable?

The missing piece here would be that it is not checked if a correct mavlink instance exists. But if you check it on mavlink you can also not check if any iridium driver is started so it is also not completely safe.

LorenzMeier replied:
This should be moved to the MAVLink module and acknowledged there. In terms of retransmission: If Iridium is not active, it should send a NACK and that will tell QGC that it is not supported.

acfloria commented:
@LorenzMeier @dagar can you look at the changes I made according to your comments.

dagar commented:
Tested?

acfloria commented:
Yes

acfloria commented:
Rebased and added a fix which avoids a rare case where the driver was stuck in a state after it received data from the ground station.

@LorenzMeier your feedback on the recent changes would be appreciated.

acfloria commented:
@LorenzMeier @dagar status?

dagar commented:
This looks good for now. In the near future I'd like to take another pass and cleanly split telemetry_status.
